java/src/com/ibm/streamsx/topology/internal/core/StreamImpl.java

```diff
@@ -362,13 +363,25 @@ public void publish(String topic) {
         this.connectTo(op, false, null);
     }
     
-    public TStream<T> parallel(int width, Routing routing) {
+    public TStream<T> parallel(final int width, Routing routing) {
+        @SuppressWarnings("serial")
+        Supplier<Integer> supplier = new Supplier<Integer>() {
+            @Override
+            public Integer get() { return width; }};
+        return parallel(supplier, routing);
+    }
+    
+    @Override
+    public TStream<T> parallel(Supplier<Integer> width, Routing routing) {
 
-        if (width <= 0) {
-            throw new IllegalStateException(
-                    "The parallel width must be greater"
-                            + " than or equal to 1.");
-        }
+        if (width == null)
+            throw new IllegalArgumentException("width");
+        Integer widthVal = width.get() != null
+                ? width.get()
+                : ((SubmissionParameter<Integer>)width).getDefaultValue();
```

Comment:
This will throw an ClassCastException is the supplier returns null and is not a submission value.